Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp

```diff
@@ -0,0 +1,537 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "WPEQtView.h"
+
+#include "WPEViewQtQuick.h"
+#include "WPEQtViewLoadRequest.h"
+#include "WPEQtViewLoadRequestPrivate.h"
+
+#include <QQuickWindow>
+#include <QSGSimpleTextureNode>
+#include <wtf/glib/GUniquePtr.h>
+
+/*!
+  \qmltype WPEView
+  \inqmlmodule org.wpewebkit.qtwpe
+  \brief A component for displaying web content.
+
+  WPEView is a component for displaying web content which is implemented using native
+  APIs on the platforms where this is available, thus it does not necessarily require
+  including a full web browser stack as part of the application.
+
+  WPEView provides an API compatible with Qt's QtWebView component. However
+  WPEView is limited to Linux platforms supporting EGL KHR extensions. WPEView
+  was successfully tested with the EGLFS and Wayland-EGL QPAs.
+*/
+WPEQtView::WPEQtView(QQuickItem* parent)
+    : QQuickItem(parent)
+{
+    connect(this, &QQuickItem::windowChanged, this, &WPEQtView::configureWindow);
+    setFlag(ItemHasContents, true);
+    setAcceptedMouseButtons(Qt::AllButtons);
+    setAcceptHoverEvents(true);
+    setAcceptTouchEvents(true);
+}
+
+WPEQtView::~WPEQtView()
+{
+    if (m_webView)
+        g_signal_handlers_disconnect_by_data(m_webView.get(), this);
+}
+
+void WPEQtView::geometryChange(const QRectF& newGeometry, const QRectF&)
+{
+    m_size = newGeometry.size().toSize();
+    if (!m_webView)
+        return;
+
+    if (auto* wpeView = webkit_web_view_get_wpe_view(m_webView.get()))
+        wpe_view_resize(wpeView, m_size.width(), m_size.height());
+}
+
+void WPEQtView::configureWindow()
+{
+    auto* win = window();
+    if (!win)
+        return;
+
+    win->setSurfaceType(QWindow::OpenGLSurface);
+
+    if (win->isSceneGraphInitialized())
+        createWebView();
+    else
+        connect(win, &QQuickWindow::sceneGraphInitialized, this, &WPEQtView::createWebView);
+}
+
+void WPEQtView::createWebView()
+{
+    RELEASE_ASSERT(!m_webView);
+
+    GUniqueOutPtr<GError> error;
+    auto* wpeDisplay = wpe_display_qtquick_new();
+    if (!wpe_display_connect(wpeDisplay, &error.outPtr())) {
+        g_warning("Failed to initialize WPE display: %s", error->message);
+        return;
+    }
+
+    auto settings = adoptGRef(webkit_settings_new_with_settings("enable-developer-extras", TRUE,
+        "enable-webgl", TRUE, "enable-mediasource", TRUE, nullptr));
+    m_webView = adoptGRef(WEBKIT_WEB_VIEW(g_object_new(WEBKIT_TYPE_WEB_VIEW,
+        "display", wpeDisplay,
+        "settings", settings.get(), nullptr)));
+
+    auto* wpeView = webkit_web_view_get_wpe_view(m_webView.get());
+    RELEASE_ASSERT(wpeView);
```

Comment:
I don't think the assert is needed, you have just created the view with a WPEDisplay.

Reply:
Ok, I'll get rid of them.